### ISSUE TITLE
TS error with v7.0.0 - Type error: Module '"../../../node_modules/markdown-to-jsx/dist"' has no exported member 'MarkdownOptions'. Did you mean to use 'import MarkdownOptions from "../../../node_modules/markdown-to-jsx/dist"' instead?

### ISSUE THREAD

**Issue by Vadorequest** [probablyup/markdown-to-jsx#332]
The TS types haven't been published to definitelyTyped `@types/markdown-to-jsx` and upgrading to v7.0.0 crashes because of some incompatibility.

I didn't encounter this issue locally, but I did encounter it when deploying to Vercel:
![image](https://user-images.githubusercontent.com/3807458/92453242-4e202b00-f1bf-11ea-8ecf-f8188d1eae11.png)

I'll rollback to 6.11.2 until newer TS types are added.

```
"markdown-to-jsx": "7.0.0",
"@types/markdown-to-jsx": "6.11.2",
```

**Comment by probablyup** [comment]
Do you happen to have third party types for this library installed?

I'm shipping them with the library itself now. `Options` is the proper name of the type I wrote.

**Comment by Vadorequest** [comment]
Ah, didn't think about that, I was still using DefinitelyTyped, so they must have clashed together. I'll give it another try!

**Comment by Vadorequest** [comment]
Okay, so indeed removing types from DefinitelyTyped helped about resolving the proper error locally.

![image](https://user-images.githubusercontent.com/3807458/92484541-18dd0280-f1ea-11ea-998d-f853b9a7cf9a.png)

But I can't use the types though, because they're not exported in the .ts file.

```ts
/**
 * markdown-to-jsx@6 is a fork of [simple-markdown v0.2.2](https://github.com/Khan/simple-markdown)
 * from Khan Academy. Thank you Khan devs for making such an awesome and extensible
 * parsing infra... without it, half of the optimizations here wouldn't be feasible. 🙏🏼
 */
import React from 'react';
declare type CreateElement = typeof React.createElement;
declare type HTMLTags = keyof JSX.IntrinsicElements;
/**
 * RequireAtLeastOne<{ ... }> <- only requires at least one key
 */
declare type RequireAtLeastOne<T, Keys extends keyof T = keyof T> = Pick<T, Exclude<keyof T, Keys>> & {
    [K in Keys]-?: Required<Pick<T, K>> & Partial<Pick<T, Exclude<Keys, K>>>;
}[Keys];
declare type Override = RequireAtLeastOne<{
    component: React.ComponentType<any>;
    props: Object;
}> | React.ComponentType<any>;
declare type Overrides = {
    [tag in HTMLTags]?: Override;
} & {
    [customComponent: string]: Override;
};
declare type Options = Partial<{
    /**
     * Ultimate control over the output of all rendered JSX.
     */
    createElement: (tag: Parameters<CreateElement>[0], props: React.Props<any>, ...children: React.ReactChild[]) => JSX.Element;
    /**
     * Disable the compiler's best-effort transcription of provided raw HTML
     * into JSX-equivalent. This is the functionality that prevents the need to
     * use `dangerouslySetInnerHTML` in React.
     */
    disableParsingRawHTML: boolean;
    /**
     * Forces the compiler to always output content with a block-level wrapper
     * (`<p>` or any block-level syntax your markdown already contains.)
     */
    forceBlock: boolean;
    /**
     * Forces the compiler to always output content with an inline wrapper (`<span>`)
     */
    forceInline: boolean;
    /**
     * Supply additional HTML entity: unicode replacement mappings.
     *
     * Pass only the inner part of the entity as the key,
     * e.g. `&le;` -> `{ "le": "\u2264" }`
     *
     * By default
     * the following entites are replaced with their unicode equivalents:
     *
     * ```
     * &amp;
     * &apos;
     * &gt;
     * &lt;
     * &nbsp;
     * &quot;
     * ```
     */
    namedCodesToUnicode: {
        [key: string]: string;
    };
    /**
     * Selectively control the output of particular HTML tags as they would be
     * emitted by the compiler.
     */
    overrides: Overrides;
    /**
     * Override normalization of non-URI-safe characters for use in generating
     * HTML IDs for anchor linking purposes.
     */
    slugify: (source: string) => string;
}>;
export declare function compiler(markdown: string, options?: Options): JSX.Element;
/**
 * A simple HOC for easy React use. Feed the markdown content as a direct child
 * and the rest is taken care of automatically.
 */
declare const Markdown: React.FC<{
    [key: string]: any;
    children: string;
    options?: Options;
}>;
export default Markdown;

```

Only `compiler` is found, because it's indeed exported. But you should use `export declare` everywhere, instead of just `declare`, so that we may import them.

**Comment by Vadorequest** [comment]
Also, the type `Override` seems too restrictive:

```ts
declare type Override = RequireAtLeastOne<{
    component: React.ComponentType<any>;
    props: Object;
}> | React.ComponentType<any>;
```

I was using a `component: 'a',` and it worked well, but now I have a TS error because `'a'` isn't assignable to ComponentType.

![image](https://user-images.githubusercontent.com/3807458/92485162-e2ec4e00-f1ea-11ea-8dd7-2774c53456c4.png)

Full code:
https://github.com/UnlyEd/next-right-now/blob/286ea44c870bf7da1a5de2b5fd33fd7825d7a867/src/components/utils/Markdown.tsx#L29

**Comment by probablyup** [comment]
a renders as an a, you don’t need to supply component unless you’re actually overriding the rendered output

**Comment by rpeach** [comment]
Workaround for Options not being exported:
```
type FunctionalComponentPropsType<T> = T extends React.FC<infer X> ? X : never;
type Options = FunctionalComponentPropsType<typeof ReactMarkdown>['options'];
```

**Comment by probablyup** [comment]
Fixed in 7.0.1 https://github.com/probablyup/markdown-to-jsx/releases/tag/7.0.1

**Comment by Vadorequest** [comment]
Thank you, here is what I did to migrate to v7 with new TS types.

https://github.com/UnlyEd/next-right-now/commit/36105610541e48094c50933c0bffa50f95708d6b
